src/main/java/com/scalar/db/storage/cosmos/DeleteStatementHandler.java

```diff
@@ -40,10 +40,15 @@ private void execute(Mutation mutation) throws CosmosException {
     CosmosMutation cosmosMutation = new CosmosMutation(mutation, metadataManager);
     cosmosMutation.checkArgument(Delete.class);
 
-    String id = cosmosMutation.getId();
-    PartitionKey partitionKey = cosmosMutation.getCosmosPartitionKey();
-    CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+    if (cosmosMutation.isPrimaryKeySpecified()) {
+      String id = cosmosMutation.getId();
+      PartitionKey partitionKey = cosmosMutation.getCosmosPartitionKey();
+      CosmosItemRequestOptions options = new CosmosItemRequestOptions();
 
-    getContainer(mutation).deleteItem(id, partitionKey, options);
+      getContainer(mutation).deleteItem(id, partitionKey, options);
+    } else {
+      // clustering key is not specified
```

Comment:
Just a question. Is it OK if clustering keys are partially specified ?
If it is OK, I think the comment should be updated like this.
```
// clustering key is not fully specified
```

Reply:
Thank you! Partially specified clustering keys are supported.
I've updated the comment.